src/k8s/pkg/k8sd/database/capi_auth.go

```diff
@@ -0,0 +1,48 @@
+package database
+
+import (
+	"context"
+	"crypto/subtle"
+	"database/sql"
+	_ "embed"
+	"fmt"
+
+	"github.com/canonical/microcluster/cluster"
+)
+
+var (
+	capiTokenStmts = map[string]int{
+		"insert-token": MustPrepareStatement("capi-auth", "insert-token.sql"),
+		"select-token": MustPrepareStatement("capi-auth", "select.sql"),
+	}
```

Comment:
There is no need to define a separate table for which only one row is ever used, and where we always check for the row ID to be 1. It is very "esoteric", and the error messages will not be useful for users to understand what is happening.

For simplicity, I would instead recommend using the cluster_configs table, with a config key like `clusterapi::auth-token` or something similar. This is functionally equivalent with what you already have, but re-uses existing constructs in the code more effectively.